[MM-19227] shift focus onto the next settings item

Willyfrog opened the pull request:
#### Summary
Once you save one of item from user account settings, focus should go to the next section so the user can edit everything in one go.

also fixed .eslintrc.json wasn't configured for jsx within .jsx files
also fixed `codeBlockOnCtrlEnter` was receiving a string while expecting a boolean.

#### Ticket Link

[MM-19227](https://mattermost.atlassian.net/browse/MM-19227)

jwilander commented on components/user_settings/advanced/user_settings_advanced.jsx:
```diff
@@ -85,6 +86,11 @@ export default class AdvancedSettingsDisplay extends React.PureComponent {
     }
 
+    componentDidMount() {
+        const prev = this.isDeactivateEnabled() ? 'deactivateAccount' : 'advancedPreviewFeatures';
```
Worth throwing `this.isDeactivateEnabled() ? 'deactivateAccount' : 'advancedPreviewFeatures'` in its own method? I see it being used in one other place

Willyfrog replied:
just trying to make explicit the intent, I can replace with its contents if you prefer that way.

jwilander replied:
0/5, I'll leave it up to you

mattermod commented:
Creating a new SpinWick test server using Mattermost Cloud.

mattermod commented:
Mattermost test server created! :tada:

Access here: https://mattermost-webapp-pr-3965.test.mattermost.cloud

| Account Type | Username | Password |
|---|---|---|
| Admin | sysadmin | Sys@dmin123 |
| User | user-1 | User-1@123 |

esethna commented:
This PR seems to break

 keyboard navigation in the Account Settings. Devin should review.

It seems you can no longer hit ENTER on an edit field and have the first field focused in the expanded setting.

See community-release:
![settings](https://user-images.githubusercontent.com/14333569/66874091-08cb8d00-ef5f-11e9-893d-5eb785c7eb94.gif)

mattermod commented:
New commit detected. SpinWick will upgrade if the updated docker image is available.

Willyfrog commented:
Added setting focus to the maximized setting.

- moved maximized setting to /components, as it now uses the store to comunicate it is closing.
- fixed tests as that broke many parts
- fixed a couple of warnings appearing during compiling

Willyfrog commented:
@jwilander @thekiiingbob I think you are both making valid points, but this change has gone too big, should we move those to a separate ticket so we can refactor the settings as every tab is different from the previous one?

mattermod commented:
Mattermost test server updated with git commit `2508d8bbd1d94af51b348047dc7968e5e559b522`.

Access here: https://mattermost-webapp-pr-3965.test.mattermost.cloud

mattermod commented:
New commit detected. SpinWick will upgrade if the updated docker image is available.

mattermod commented:
Mattermost test server updated with git commit `ee130c4a708c807ae2a8784580a895d6a1963ac8`.

Access here: https://mattermost-webapp-pr-3965.test.mattermost.cloud

esethna commented:
The first focused item doesn't seem to be visually focused, even though it is. Below "Desktop Notifications" is seemingly skipped by the visual focus, but I can still tab to it. Same happens after saving a setting now

![tabbing](https://user-images.githubusercontent.com/14333569/67132836-fb144280-f1be-11e9-8a21-25e13a6e3adf.gif)

esethna commented:
Is this PR necessary? Seems like we get the expected behaviour from this PR? ie you can keep tabbing through additional settings after hitting save.

https://github.com/mattermost/mattermost-webapp/pull/3946

jwilander commented:
@Willyfrog My only concern about separate tickets is that they will get lost in the backlog and only completed after we run into a bunch of bugs like this. I think this is a case where we should be focusing on high quality over rushing something in. I like @thekiiingbob's suggestion of a cypress test that would be able to check if something breaks. With that I will be much happier merging this as is

Willyfrog commented:
Thanks @jwilander ! It's a good point, and I'll try hard to make sure it doesn't fall into oblivion. But if we start a refactor on this section, I think we should do it of the whole `Account Settings` section, as every part is a bit different from the other and we should standardize to make sure we don't run into easy to break code, which is what created this bug originally (it's a regression from 5.10). Unfortunately, it is a big enough task that I don't think I should take it on my current sprint (this task already grew way too big with 47 files changed) and having it grow bigger will definitely generate bigger problems. Anyways I went ahead and created a ticket for [the whole refactor](https://mattermost.atlassian.net/browse/MM-19576) with a special note to ensure that we don't miss your suggestion :)

On the other hand, I'll do bob's suggestion of a cypress test, as that will help both now and when we do the refactor, ensuring less errors.

jwilander commented:
Sounds good to me 👍

Willyfrog commented:
@esethna I added @devinbinnie changes for the keyboard navigation

mattermod commented:
Test server destroyed

mattermod commented:
Creating a new SpinWick test server using Mattermost Cloud.

mattermod commented:
Test server creation failed. See the logs for more information.

mattermod commented:
Test server destroyed

esethna commented:
Thanks @Willyfrog, can you help get the test server back up? ^

mattermod commented:
Creating a new SpinWick test server using Mattermost Cloud.

mattermod commented:
Mattermost test server created! :tada:

Access here: https://mattermost-webapp-pr-3965.test.mattermost.cloud

| Account Type | Username | Password |
|---|---|---|
| Admin | sysadmin | Sys@dmin123 |
| User | user-1 | User-1@123 |

mattermod commented:
Mattermost test server updated with git commit `6b3a0abd3b855949125b4768e2da15f0dda091fe`.

Access here: https://mattermost-webapp-pr-3965.test.mattermost.cloud

mattermod commented:
New commit detected. SpinWick will upgrade if the updated docker image is available.

mattermod commented:
New commit detected. SpinWick will upgrade if the updated docker image is available.

mattermod commented:
Test server creation failed. See the logs for more information.

mattermod commented:
Mattermost test server updated with git commit `3fbc865432ef8617360068568f0fe44029d4297c`.

Access here: https://mattermost-webapp-pr-3965.test.mattermost.cloud

Willyfrog commented:
added focus for radio input

thekiiingbob commented on e2e/cypress/integration/account_settings/general/focus_spec.js:
```diff
@@ -0,0 +1,56 @@
+        cy.toAccountSettingsModal('user-1');
+    });
+
+    it('focus is in editing the full name', () => {
```
The name of the test is the same across all the files. They should be different, right?

Willyfrog replied:
since they are in different folders, I thought it would be ok. I'll change them

esethna commented:
@Willyfrog i think this piece of feedback may have been missed in the iteration? 
![image](https://user-images.githubusercontent.com/14333569/67524093-e9bab280-f664-11e9-9252-bc136ba08f1e.png)

mattermod commented:
New commit detected. SpinWick will upgrade if the updated docker image is available.

Willyfrog commented:
@esethna I've fixed 1 and 2, I'm not really sure I understand the third point, can we discuss in a zoom call?

mattermod commented:
Mattermost test server updated with git commit `aab0bb56e64598effb82bc67b03e159356979e55`.

Access here: https://mattermost-webapp-pr-3965.test.mattermost.cloud

mattermod commented:
Test server destroyed

esethna commented:
@Willyfrog should we close this PR and reassign/reopen the ticket or is it still in progress?

Willyfrog commented:
@esethna since these parts are all moved to ts, we have to recreate the whole thing again. Closing, no need to create new tickets, the original one is still valid, but we need to port most of this work to the new language
